### PR TITLE
reflow fixup for README.md / cmark

### LINKED ISSUE #126: reflow display is messed up for multilines / pointless selection
**Describe the bug**

Reflow is _always_ a multiline case with only one option. #25 is yet to be implemented, so the display is distorted for a simple yes/no response.

**To Reproduce**

Steps to reproduce the behaviour:

1. `cargo run -- reflow demo`
2. observe

**Expected behavior**

Do not ask whether to reflow, just reflow all of it.

Alternative: impl #25 and do not show a custom replacement option, it's just too error prone. Also make sure to show the full lines with some extra context and highlighted changes instead of only the replacement.

**Pplease complete the following information:**
 - System:  Fedora 33
 - Obtained:  git 
 - Version:  cargo-spellcheck 0.4.7-alpha.0

### LINKED ISSUE #128: crash on common mark README reflow
**Describe the bug**

**To Reproduce**

Steps to reproduce the behaviour:

1. A file containing `RUST_BACKTRACE=1 cargo run -- reflow -vvvv README.md; echo $?`
2. ...

```
    Finished dev [unoptimized + debuginfo] target(s) in 0.04s
     Running `target/debug/cargo-spellcheck reflow -vvvv README.md`
[2020-11-10T12:11:04Z INFO  cargo_spellcheck] Using configuration file /media/supersonic1t/projects/cargo-spellcheck/.config/spellcheck.toml
[2020-11-10T12:11:04Z DEBUG cargo_spellcheck] Executing: Reflow with Config { hunspell: Some(HunspellConfig { lang: Some("en_US"), search_dirs: SearchDirs(["/media/supersonic1t/projects/cargo-spellcheck/.config", "/usr/share/myspell"]), extra_dictionaries: ["/media/supersonic1t/projects/cargo-spellcheck/.config/lingo.dic"], quirks: Quirks { transform_regex: [WrappedRegex(^'([^\s])'$), WrappedRegex(^[0-9]+x$), WrappedRegex(^\#[0-9]+$), WrappedRegex(^[0-9]+$), WrappedRegex(^.+\+$)], allow_concatenation: true, allow_dashes: false } }), languagetool: None, reflow: Some(ReflowConfig { max_line_length: 80 }) }
[2020-11-10T12:11:04Z DEBUG cargo_spellcheck::traverse] Running on inputs ["README.md"] / recursive=false
[2020-11-10T12:11:04Z DEBUG cargo_spellcheck::traverse] Processing README.md -> /media/supersonic1t/projects/cargo-spellcheck/README.md
[2020-11-10T12:11:04Z DEBUG cargo_spellcheck::traverse] Running on absolute dirs ["/media/supersonic1t/projects/cargo-spellcheck/README.md"] 
[2020-11-10T12:11:04Z DEBUG cargo_spellcheck::traverse] Found a total of 1 files to check 
[2020-11-10T12:11:04Z DEBUG cargo_spellcheck::reflow] Reflowing CommonMarkFile("/media/supersonic1t/projects/cargo-spellcheck/README.md")
[src/reflow/mod.rs:459] parent = 21..87
[src/reflow/mod.rs:459] &cover.start = 21
[src/reflow/mod.rs:460] parent = 21..87
[src/reflow/mod.rs:460] &(cover.end - 1) = 86
[src/reflow/mod.rs:459] parent = 21..87
[src/reflow/mod.rs:459] &cover.start = 20
thread 'main' panicked at 'assertion failed: dbg!(parent).contains(dbg!(& cover . start))', src/reflow/mod.rs:459:29
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.46/src/backtrace/libunwind.rs:86
   1: backtrace::backtrace::trace_unsynchronized
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.46/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print_fmt
             at src/libstd/sys_common/backtrace.rs:78
   3: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
             at src/libstd/sys_common/backtrace.rs:59
   4: core::fmt::write
             at src/libcore/fmt/mod.rs:1076
   5: std::io::Write::write_fmt
             at src/libstd/io/mod.rs:1537
   6: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:62
   7: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:49
   8: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:198
   9: std::panicking::default_hook
             at src/libstd/panicking.rs:217
  10: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:526
  11: std::panicking::begin_panic
             at /home/bernhard/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/panicking.rs:456
  12: cargo_spellcheck::reflow::reflow
             at src/reflow/mod.rs:459
  13: <cargo_spellcheck::reflow::Reflow as cargo_spellcheck::checker::Checker>::check::{{closure}}
             at src/reflow/mod.rs:40
  14: <rayon::iter::try_fold::TryFoldFolder<C,U,F> as rayon::iter::plumbing::Folder<T>>::consume
             at /home/bernhard/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-1.5.0/src/iter/try_fold.rs:147
  15: rayon::iter::plumbing::Folder::consume_iter
             at /home/bernhard/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-1.5.0/src/iter/plumbing/mod.rs:179
  16: <rayon::iter::map::MapFolder<C,F> as rayon::iter::plumbing::Folder<T>>::consume_iter
             at /home/bernhard/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-1.5.0/src/iter/map.rs:248
  17: rayon::iter::plumbing::Producer::fold_with
             at /home/bernhard/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-1.5.0/src/iter/plumbing/mod.rs:110
  18: rayon::iter::plumbing::bridge_producer_consumer::helper
             at /home/bernhard/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-1.5.0/src/iter/plumbing/mod.rs:438
  19: rayon::iter::plumbing::bridge_producer_consumer
             at /home/bernhard/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-1.5.0/src/iter/plumbing/mod.rs:397
  20: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback
             at /home/bernhard/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-1.5.0/src/iter/plumbing/mod.rs:373
  21: <rayon::slice::Iter<T> as rayon::iter::IndexedParallelIterator>::with_producer
             at /home/bernhard/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-1.5.0/src/slice/mod.rs:543
  22: rayon::iter::plumbing::bridge
             at /home/bernhard/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-1.5.0/src/iter/plumbing/mod.rs:357
  23: <rayon::slice::Iter<T> as rayon::iter::ParallelIterator>::drive_unindexed
             at /home/bernhard/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-1.5.0/src/slice/mod.rs:519
  24: <rayon::iter::map::Map<I,F> as rayon::iter::ParallelIterator>::drive_unindexed
             at /home/bernhard/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-1.5.0/src/iter/map.rs:49
  25: <indexmap::rayon::map::ParIter<K,V> as rayon::iter::ParallelIterator>::drive_unindexed
             at /home/bernhard/.cargo/registry/src/github.com-1ecc6299db9ec823/indexmap-1.6.0/src/rayon/mod.rs:20
  26: <rayon::iter::try_fold::TryFold<I,U,ID,F> as rayon::iter::ParallelIterator>::drive_unindexed
             at /home/bernhard/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-1.5.0/src/iter/try_fold.rs:64
  27: rayon::iter::try_reduce::try_reduce
             at /home/bernhard/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-1.5.0/src/iter/try_reduce.rs:20
  28: rayon::iter::ParallelIterator::try_reduce
             at /home/bernhard/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-1.5.0/src/iter/mod.rs:1070
  29: <cargo_spellcheck::reflow::Reflow as cargo_spellcheck::checker::Checker>::check
             at src/reflow/mod.rs:34
  30: cargo_spellcheck::run
             at src/main.rs:384
  31: cargo_spellcheck::main
             at src/main.rs:400
  32: std::rt::lang_start::{{closure}}
             at /home/bernhard/.rustup/toolchains/1.46.0-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/rt.rs:67
  33: std::rt::lang_start_internal::{{closure}}
             at src/libstd/rt.rs:52
  34: std::panicking::try::do_call
             at src/libstd/panicking.rs:348
  35: std::panicking::try
             at src/libstd/panicking.rs:325
  36: std::panic::catch_unwind
             at src/libstd/panic.rs:394
  37: std::rt::lang_start_internal
             at src/libstd/rt.rs:51
  38: std::rt::lang_start
             at /home/bernhard/.rustup/toolchains/1.46.0-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/rt.rs:67
  39: main
  40: __libc_start_main
  41: _start
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
101
```

**Expected behavior**

No crash, exit code 0.

**Pplease complete the following information:**
 - System:  Fedora 33
 - Obtained:  git
 - Version: 

**Additional context**

### PR DESCRIPTION
**Pull request by drahnr**
## What does this PR accomplish?

 * 🩹 Bug Fix
 * 🪣 Misc

Closes #128 .
Closes #126 .

## Changes proposed by this PR:

Add run checks to CI, assure it's not going to break again.
Addition debug logs for reflow.
Assorted debug adjustments (only singlethreaded checker execution).

## 📜 Checklist

 * [x] Works on the `./demo` sub directory
 * [x] Test coverage is excellent and passes
 * [x] Documentation is thorough

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -140,29 +142,35 @@ Available checker support
...
 ##### Mac OS X
-```
+
+```sh
 brew install llvm
 ```
 
-The environment variable `LLVM_CONFIG_PATH` needs to point to `llvm-config`, to do so:
+The environment variable `LLVM_CONFIG_PATH` needs to point to `llvm-config`, to
+do so:
```
Comment (KuabeM) [inline README.md:167]: DId someone use reflow on the README? :heart:

File: src/documentation/chunk.rs [hunk 2]
```diff
@@ -204,7 +204,7 @@ impl CheckableChunk {
                 let mut iter = s.chars().enumerate().peekable();
                 let mut started = true;
                 'w: while let Some((idx, c)) = iter.next() {
-                    trace!(target: "find_spans", "char[{}]: {}", idx, c);
+                    // trace!(target: "find_spans", "char[{}]: {}", idx, c);
```
Comment (KuabeM) [inline src/documentation/chunk.rs:207]: Could be removed?

File: src/reflow/mod.rs [hunk 3]
```diff
@@ -286,27 +286,37 @@ impl<'s> Indentation<'s> {
 /// Note: Leading spaces are skipped by the markdown parser,
 /// which implies for `///` and `//!`, the paragraph for the first
 /// line starting right after `/// ` (note the space here).
+///
+///
+/// Returns: TODO FIXME ??
```
Comment (KuabeM) [inline src/reflow/mod.rs:291]:
```suggestion
/// Returns: end of processed range and Suggestion, if reflow happened
```

File: README.md (hunk 1 above)
Comment (drahnr) [reply README.md:167]: Yeah, but I did not realize it fucked up...

File: src/reflow/mod.rs (hunk 3 above)
Comment (drahnr) [reply src/reflow/mod.rs:291]: In bytes I assume.

File: src/documentation/chunk.rs (hunk 2 above)
Comment (drahnr) [reply src/documentation/chunk.rs:207]:
```suggestion
```

File: README.md (hunk 1 above)
Comment (drahnr) [reply README.md:167]: And it worked nicely unlike my first comment suggested (phone renders of diffs are no good for that)

**Review by KuabeM — APPROVED** [review]

**Pull request merged**